usr/lib/pkcs11/soft_stdll/soft_specific.c

```diff
@@ -2166,6 +2217,13 @@ CK_RV token_specific_sha(STDLL_TokData_t *tokdata, DIGEST_CONTEXT *ctx,
     if (!rc)
         goto error;
 
+    switch (ctx->mech.mechanism) {
+    case CKM_SHA512_224:
+    case CKM_SHA512_256:
+        memcpy(orig_out_data, temp_out_data, hlen);
```

Comment:
You may want to erase the non-output bytes from memory here: OPENSSL_cleanse(temp_out_data, sizeof(temp_out_data))

Reply:
fixed